looper/cli_pydantic.py

```diff
@@ -43,10 +42,46 @@
     is_registry_path,
     read_looper_config_file,
     read_looper_dotfile,
+    initiate_looper_config,
+    init_generic_pipeline,
+    read_yaml_file,
 )
 
+from typing import List, Tuple
 
-def run_looper(args: TopLevelParser, parser: ArgumentParser):
+
+def opt_attr_pair(name: str) -> Tuple[str, str]:
+    return f"--{name}", name.replace("-", "_")
+
+
+def validate_post_parse(args: argparse.Namespace) -> List[str]:
+    problems = []
+    used_exclusives = [
+        opt
+        for opt, attr in map(
+            opt_attr_pair,
+            [
+                "skip",
+                "limit",
+                SAMPLE_EXCLUSION_OPTNAME,
+                SAMPLE_INCLUSION_OPTNAME,
+            ],
+        )
+        # Depending on the subcommand used, the above options might either be in
+        # the top-level namespace or in the subcommand namespace (the latter due
+        # to a `modify_args_namespace()`)
+        if getattr(
+            args, attr, None
+        )  # or (getattr(args.run, attr, None) if hasattr(args, "run") else False)
+    ]
+    if len(used_exclusives) > 1:
+        problems.append(
+            f"Used multiple mutually exclusive options: {', '.join(used_exclusives)}"
+        )
+    return problems
+
+
+def run_looper(args: TopLevelParser, parser: ArgumentParser, test_args=None):
```

Comment:
I think we have ability to run looper as a python library, right? In which case, should this function be renamed to `run_looper_via_cli` to distinguish? Because I think this one is not the one you'd use from interactive Python, right?